src/commonControls/numberControl/NumberControl.ts

```diff
@@ -110,17 +116,16 @@ export interface NumberControlProps extends ControlProps {
      *  - the Control will take initiative to explicitly confirm the value with a yes/no
      *    question.
      */
-    confirmationRequired?: boolean | NumberConfirmationRequireFunction;
+    confirmationRequired?: boolean | ((state: NumberControlState, input: ControlInput) => boolean);
 
-    // TODO: feature. generalize this to "valueToSuggest"
-    /**
-     * List of value-pairs that are known to be frequently misunderstood by NLU
+    /*
+     * Function that determines the value which is known to be frequently misunderstood by NLU
      *
      * Control behavior:
-     * - If the user disaffirms one value of a pair, the other will be
-     *   suggested.
+     * - If the user disaffirms a value, return value of `mostLikelyMisunderstandings`
+     * function is suggested back to the user.
      */
-    ambiguousPairs?: Array<[number, number]>;
+    mostLikelyMisunderstandings?: (value: number, input: ControlInput) => number;
```

Comment:
I think singular is more correct as the function returns one number. -> `mostLikelyMisunderstanding`